File: android/src/main/kotlin/com/gstory/flutter_unionad/FlutterUnionadEventPlugin.kt
package com.gstory.flutter_unionad

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel

/**
 * @Description:
 * @Author: gstory0404@gmail
 * @CreateDate: 2020/8/19 18:52
 */
class FlutterUnionadEventPlugin : FlutterPlugin, EventChannel.StreamHandler {
    private var channelName = "com.gstory.flutter_unionad/adevent";

    companion object {
        private var eventChannel: EventChannel? = null

        private var eventSink: EventChannel.EventSink? = null

        private var context: Context? = null

        fun sendContent(content:Map<String,Any>) {
            eventSink?.success(content);
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        eventChannel = EventChannel(binding.binaryMessenger, channelName)
        eventChannel!!.setStreamHandler(this)
        context = binding.applicationContext
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        eventChannel = null
        eventChannel!!.setStreamHandler(null)
    }
}

File: ios/Classes/FlutterUnionadEventPlugin.swift
//
//  FlutterUnionadEventPlugin.swift
//  flutter_unionad
//
//  Created by gstory0404@gmail on 2020/9/3.
//

import Foundation
import Flutter

public class FlutterUnionadEnentPlugin : NSObject, FlutterStreamHandler{
    
    private var eventSink:FlutterEventSink? = nil
    
    init(_ registrar: FlutterPluginRegistrar){
        super.init()
        let eventChannel = FlutterEventChannel.init(name: FlutterUnionadConfig.event.adevent, binaryMessenger:registrar.messenger())
        eventChannel.setStreamHandler((self as FlutterStreamHandler & NSObjectProtocol))
    }
    
    
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
    
    //发送event
    public func sendEvent(event:NSDictionary) {
        eventSink?(event)
    }

}
